pkg/kvstore/store/syncstore.go

```diff
@@ -80,12 +80,19 @@ type wqSyncStore struct {
 	syncedMetric prometheus.Gauge
 }
 
-type syncCanary struct{ skipCallbacks bool }
+const (
+	syncCanary                  = "_sync_canary"
+	syncCanaryWithSkipCallBacks = syncCanary + "_skip_callbacks"
```

Comment:
Is it safe to assume that these will never be actual keys/could clash?

Reply:
Here its harder to assert, since the keys that go into this are not always from a Kubernetes name. For example `IPIdentityPair` objects are also stored in this queue, their keys being IP CIDRs (also would not clash). So its difficult to guarantee no clash can ever occur, specifically if the key may contain user controllable content. Future users of the store would have to guard for that, which isn't likely to be known.

So I would suggest finding some alternative way to signal. Since both enqueue point and dequeue points are internal to the syncstore, I suggest wrapping the string in a struct and adding flags to that struct.

Reply:
That seems like a good idea, used a struct instead to make this safer